### PR TITLE
i8042 - initial implementation

### PR DESCRIPTION
**Pull request by lauralt**

### CODE REVIEW

File: src/i8042.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+//! Provides emulation for a super minimal i8042 controller.
+//!
+//! This emulates just the CPU reset command.
+
+use std::fmt::{self, Display, Formatter};
+use std::{io, result};
+use vmm_sys_util::eventfd::EventFd;
+
+// Offset of the status port (port 0x64).
+const OFS_STATUS: u8 = 4;
```
Comment (alexandruag) [inline src/i8042.rs:17]: Maybe it's a bit clearer if we explicitly state in the comment that offset 4 represents the status register when reading, and the command register when writing (i.e. when sending the reset command).

File: src/i8042.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /// Handles a read request from the driver at `_offset` offset from the
+    /// base I/O address.
+    ///
+    /// Returns the read value, which at this moment is 0x00, since we're not
+    /// interested in an i8042 operation other than CPU reset.
+    ///
+    /// # Arguments
+    /// * `_offset` - The offset that will be added to the base address
+    ///              for writing to a specific register.
+    pub fn read(&mut self, _offset: u8) -> u8 {
```
Comment (alexandruag) [inline src/i8042.rs:69]: Do reads/writes always happen at 1 byte granularity?

File: src/i8042.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        return 0x00;
```
Comment (alexandruag) [inline src/i8042.rs:70]: Looks like `clippy` is complaining that we're not just saying `0x00` here instead of `return 0x00;`.

**Review by alexandruag — COMMENTED** [review]
Looks good, just left some small comments.

File: src/i8042.rs (hunk 2 above)
Comment (lauralt) [reply src/i8042.rs:69]: From my limited understanding, yes.

File: README.md [hunk 4]
```diff
@@ -3,7 +3,8 @@
 
 The `vm-superio` crate provides emulation for legacy devices. For now, it offers
 this support only for the
-[Linux serial console](https://en.wikipedia.org/wiki/Linux_console).
+[Linux serial console](https://en.wikipedia.org/wiki/Linux_console) and a minimal
+[i8042 PS/2 Controller](https://web.archive.org/web/20200602023256/https://wiki.osdev.org/%228042%22_PS/2_Controller).
```
Comment (aghecenco) [inline README.md:7]: This link looks odd. Does https://wiki.osdev.org/%228042%22_PS/2_Controller do the trick? Same for line 54.

File: coverage_config_x86_64.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "coverage_score": 100,
+  "coverage_score": 94.5,
```
Comment (aghecenco) [inline coverage_config_x86_64.json:2]: Oh noes

File: src/i8042.rs (hunk 2 above)
Comment (aghecenco) [reply src/i8042.rs:72]: Looks like it: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/input/serio/i8042-io.h#n58

File: coverage_config_x86_64.json (hunk 5 above)
Comment (andreeaflorescu) [reply coverage_config_x86_64.json:2]: :broken_heart:

File: README.md (hunk 4 above)
Comment (lauralt) [reply README.md:7]: Yup, thanks.

File: coverage_config_x86_64.json (hunk 5 above)
Comment (lauralt) [reply coverage_config_x86_64.json:2]: It's the Display thing... :( And we don't like coverage tests :(.

**Review by alexandruag — APPROVED** [review]

**Review by aghecenco — APPROVED** [review]

**Pull request merged**